pkgs/by-name/pk/pkcs11-provider/package.nix

```diff
@@ -0,0 +1,47 @@
+{ lib, stdenv, fetchFromGitHub, runCommandLocal
+, openssl, nss, p11-kit
+, opensc, gnutls, expect
+, autoconf, autoconf-archive, automake, pkg-config, libtool
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pkcs11-provider";
+  version = "0.3";
+
+  src = fetchFromGitHub {
+    owner = "latchset";
+    repo = "pkcs11-provider";
+    rev = "v${version}";
+    hash = "sha256-jEQYsINRZ7bi2UqOXUUmGpm+1h+1qBNe18KvfAw2JzU=";
+  };
+
+  buildInputs = [ openssl nss p11-kit ];
+  nativeBuildInputs = [ autoconf autoconf-archive automake pkg-config libtool ];
+
+  # don't add SoftHSM to here: https://github.com/openssl/openssl/issues/22508
+  nativeCheckInputs = [ p11-kit.bin opensc nss.tools gnutls openssl.bin expect ];
+
+  postPatch = ''
+    patchShebangs .
+
+    # Makefile redirects to logfiles; make sure we can catch them.
+    for name in softokn softhsm; do
+      ln -s /dev/stderr tests/setup-$name.log
+    done
+  '';
+
+  preConfigure = "autoreconf -fi";
+
+  enableParallelBuilding = true;
+  enableParallelInstalling = false;
```

Comment:
Is this not reproducible?

Reply:
I'll add a comment; it frequently dies due to a race.